--- Module-Build-0.21/lib/Module/Build/Base.pm.orig	2003-10-16 03:25:18.000000000 +0200
+++ Module-Build-0.21/lib/Module/Build/Base.pm	2003-10-25 01:40:41.217035488 +0200
@@ -889,7 +889,7 @@
 
   # Extract our 'properties' from $cmd_args, the rest are put in 'args'.
   foreach my $key (keys %args) {
-    my $add_to = $self->valid_property($key) ? $self->{properties} : $self->{args};
+    my $add_to = ($key eq 'config') ? $self : ($self->valid_property($key) ? $self->{properties} : $self->{args});
 
     if ($additive{$key}) {
       $add_to->{$key}{$_} = $args{$key}{$_} foreach keys %{$args{$key}};
