diff -urNp -x '*.orig' Module-Build-0.4231.org/lib/Module/Build/Base.pm Module-Build-0.4231/lib/Module/Build/Base.pm
--- Module-Build-0.4231.org/lib/Module/Build/Base.pm	2020-01-29 08:51:23.000000000 +0100
+++ Module-Build-0.4231/lib/Module/Build/Base.pm	2022-02-21 00:14:52.613086898 +0100
@@ -2978,7 +2978,7 @@ sub process_script_files {
 
   foreach my $file (sort keys %$files) {
     my $result = $self->copy_if_modified($file, $script_dir, 'flatten') or next;
-    $self->fix_shebang_line($result) unless $self->is_vmsish;
+#   $self->fix_shebang_line($result) unless $self->is_vmsish;
     $self->make_executable($result);
   }
 }
diff -urNp -x '*.orig' Module-Build-0.4231.org/lib/Module/Build/Platform/Unix.pm Module-Build-0.4231/lib/Module/Build/Platform/Unix.pm
--- Module-Build-0.4231.org/lib/Module/Build/Platform/Unix.pm	2020-01-29 08:51:23.000000000 +0100
+++ Module-Build-0.4231/lib/Module/Build/Platform/Unix.pm	2022-02-21 00:14:52.612086896 +0100
@@ -5,6 +5,7 @@ use warnings;
 our $VERSION = '0.4231';
 $VERSION = eval $VERSION;
 use Module::Build::Base;
+use Config;
 
 our @ISA = qw(Module::Build::Base);
 
@@ -19,7 +20,11 @@ sub is_executable {
   return +(stat $file)[2] & 0100;
 }
 
-sub _startperl { "#! " . shift()->perl }
+sub _startperl {
+  exists $Config{startperl} && $Config{startperl}
+    ? $Config{startperl}
+    : "#! " . shift()->perl
+}
 
 sub _construct {
   my $self = shift()->SUPER::_construct(@_);
diff -urNp -x '*.orig' Module-Build-0.4231.org/t/runthrough.t Module-Build-0.4231/t/runthrough.t
--- Module-Build-0.4231.org/t/runthrough.t	2020-01-29 08:51:23.000000000 +0100
+++ Module-Build-0.4231/t/runthrough.t	2022-02-21 00:14:52.612086896 +0100
@@ -25,7 +25,7 @@ $dist->add_file( 'MANIFEST.SKIP', <<'---
 ^MYMETA.yml$
 ---
 $dist->add_file( 'script', <<'---' );
-#!perl -w
+#!/usr/bin/perl -w
 print "Hello, World!\n";
 ---
 $dist->add_file( 'lib/Simple/Script.PL', <<'---' );
